### PR TITLE
Pulls out CT mode into open source.

### PR DESCRIPTION
**Pull request by SirVer**
CT mode sets the near and far plane really close to each other. This allows to look at a slice of the point cloud directly in front of the observer and has been useful for internal debugging at Lyft.

This open sources the implementation, but does not mention the mode in README.md, since it seems very specific.

### CODE REVIEW

File: sdl_viewer/src/camera.rs [hunk 1]
```diff
@@ -50,6 +50,9 @@ pub struct Camera {
     pub turning_up: bool,
     pub width: i32,
     pub height: i32,
+    pub ct_mode: bool,
```
Comment (feuerste) [inline sdl_viewer/src/camera.rs]: How about defining a `ct` struct bundling the settings together?

File: sdl_viewer/src/camera.rs [hunk 2]
```diff
@@ -131,18 +153,35 @@ impl Camera {
...
+    }
+
+    pub fn update_viewport(&mut self, gl: &opengl::Gl) {
         self.projection_matrix = Matrix4::from(PerspectiveFov {
             fovy: Rad::from(Deg(45.)),
-            aspect: width as f32 / height as f32,
-            near: 0.1,
-            far: 10000.0,
+            aspect: self.width as f32 / self.height as f32,
+            near: if self.ct_mode {
```
Comment (feuerste) [inline sdl_viewer/src/camera.rs]: How about declaring
```rust
let (near, far) = if self.ct_mod {
    (self.near_plane_ct, self.far_plane_ct)
} else {
    (NEAR_PLANE, FAR_PLANE)
};
```
outside and change here to only `near` and `far`.

**Comment by SirVer** [comment]
All done. PTAL.

**Comment by feuerste** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user feuerste. Merge queue now has a length of 1.

**Pull request merged**